### PR TITLE
7731 bye bye underscore => _.each

### PR DESCRIPTION
**Pull request by nathanwoulfe**
Changes replace `_.each` with native `.forEach` in directives (components) and resources - I've sanity checked these locally, all seems to be fine. This is related to #7731 but does not yet complete the _.each removal (figured it's better to have multiple smaller PRs rather than one big one with 50 files modified).

Most are like-for-like replacements, with the exception of some like the changes in umbsearch.directive.js where the value is a dictionary not an array, so uses `Object.keys().forEach()` because the value is not directly iterable.

Made related syntax changes to replace some indexOf calls with includes (easier to read `x.includes(y)` compared to `x.indexOf(y) !== -1`), and converted a few functions to single-line fat-arrows where it made sense to be terser.

### CODE REVIEW

**Comment by nul800sebastiaan** [comment]
Looks good, thanks @nathanwoulfe! 👍

File: src/Umbraco.Web.UI.Client/src/common/services/search.service.js [hunk 1]
```diff
@@ -119,7 +113,7 @@ angular.module('umbraco.services')
 
                 return entityResource.searchAll(args.term, args.canceler).then(function (data) {
 
-                    _.each(data, function (resultByType) {
+                    data.forEach(resultByType => {
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/common/services/search.service.js:116]: @nathanwoulfe Uh oh! This one seems to have broken! 

image


Little help please? 🙏

Comment (nul800sebastiaan) [reply src/Umbraco.Web.UI.Client/src/common/services/search.service.js:116]: Reproduce by trying to search in the backoffice:

image

Comment (nathanwoulfe) [reply src/Umbraco.Web.UI.Client/src/common/services/search.service.js:116]: Hmm. I thought I'd sorted that... I'm guessing data isn't an array, but an object/dictionary. I'll grab the laptop...

**Pull request merged**